### DISCUSSION TITLE
Multiple #[WithCast(SomeCast::class)]

### DISCUSSION THREAD

**Discussion by mynetx** [spatie/laravel-data#611] [Q&A] [answered]
I'm writing a file converter, and for some fields I need to apply multiple casts. How can I do this?

```php
        #[WithCast(HtmlToMarkdownCast::class), WithCast(HtmlTagsCleanerCast::class)]
        public string $content,
```

This wouldn't work.

```php
        #[WithCast(HtmlToMarkdownCast::class)]
        #[WithCast(HtmlTagsCleanerCast::class)]
        public string $content,
```

This also wouldn't work.

What's the correct approach?

**Comment by tborealis** [comment, marked as answer]
I don't think the casts are designed to do this. How about passing a flag in to the HtmlToMarkdownCast that tells it to clean the HTML?

**Reply by rubenvanassche** [reply]
Yeah that looks the best solution to me!

**Reply by mynetx** [reply]
Thank you guys! That worked for me. :)
